### DISCUSSION TITLE
Is there any way to retrieve a reverse-related field with the API?

### DISCUSSION THREAD

**Discussion by hsb-tonmoy** [directus/directus#15227] [Archive]
I feel like I didn't word the question properly. So, basically I have a collection named Articles and a collection named Stats which has a M2O relation field with Articles. The Stats collection has fields such as. views, rating, source etc.

Is there any way to retrieve fields from Stats when I'm querying Articles? Django has this feature where you define a "related_field" name and you can do a reverse-relation query with that name.

The reason I created a separate collection is because I don't want to update my Articles every time I need to increase a view.

A simpler example would be querying a Category and retrieving all the articles in that category.

**Comment by br41nslug** [comment]
For that query to work you will need to add an O2M relation so Directus knows there is a relation (o2m is an alias without an actual database column and you can hide it on the interface if you don't need it beyond querying). When you've added that you can traverse it in a query like any other relation 😄

**Comment by Nitwel** [comment]
Yes, we have that even though it is not documented.
You can use `$FOLLOW(collection,field)` to follow any relation in reverse as it is normally not possible. Though note that for following along the direction of relations you can do as brainslug mentioned.

https://github.com/directus/directus/blob/ef0a6eb9655b5a334ba15a730035ef1de595c288/app/src/interfaces/list-m2m/list-m2m.vue#L507-L517 for how we used it.

**Reply by br41nslug** [reply]
Ah good one i had absolutely forgotten about the existence of `$FOLLOW(...)` 😮

**Reply by rijkvanzanten** [reply]
While you're absolutely right, I _think_ the OP is actually referring to "just" a O2M field as per @br41nslug's answer above 🙂

**Reply by Nitwel** [reply]
Exactly what I thought too, the naming ~~is~~ was just a bit missleading.

**Reply by rokdd** [reply]
Is it possible to filter by `$FOLLOW` when querying a m2m field per REST API? I just tried around the parameters and nothing suceeded? So I made a field "parent_of" and I want to know the children of a item: `?children=$FOLLOW(<same collection>,parent_of)`

**Reply by themegabyte** [reply]
Hey @rokdd I hope your issue was solved. If not, I just dug deeper into $FOLLOW and maybe able to help you out and learn more. Let me know, cheers.
